webui/ocw/lib/azure.py

```diff
@@ -44,21 +40,21 @@ def check_credentials(self):
         raise AuthenticationError("Invalid Azure credentials")
 
     def sp_credentials(self):
-        if (self.__sp_credentials is None):
+        if (self.__sp_credentials is None or self.__credentials.isExpired()):
```

Comment:
Same problem as in EC2(). The `self.__compute_mgmt_client` and `self.__resource_mgmt_client` will not renewed.